### ISSUE TITLE
Should identity and viewerSpace be merged?

### ISSUE THREAD

**Issue by Manishearth** [immersive-web/webxr#623]
Brought up by @NellWaliczek in https://github.com/immersive-web/webxr/pull/621#discussion_r280544401

See also: https://github.com/immersive-web/webxr/issues/478

`identity` reference spaces and `session.viewerSpace` are both reference spaces with a native origin that matches the viewer/headset pose.

The differences are:

 - `viewerSpace` is not a reference space and cannot be fed to `getViewerPose()`. It also does not have an origin offset
 - `identity` is (currently) restricted as to what spaces it can be related against in `getPose()`, currently you can only relate it against `viewerSpace` and `screen-space` targetRaySpace`s. This is for privacy/permissions reasons.

It should be possible to merge them. [I don't think it is necessary to restrict `identity`'s ability to relate to other spaces for _permissions_ reasons:](https://github.com/immersive-web/webxr/issues/478#issuecomment-483840400) the only way relating an identity space with other spaces can leak pose information is if you have another space that is not tied to the user's pose, and we can permissions-gate those anyway. Any information someone wants to leak from `identity` can be leaked through `viewerSpace` anyway, so whatever we design for permissions needs to be able to handle both.

One non-permissions concern of merging the two is that the concept of an identity space makes less sense in non-screen-space experiences, it's only an identity when related to the viewer pose (ignoring origin offsets).

A potential way of fixing that is to replace "identity" with a "viewer" reference space. This space can be related to everything, and it now has an origin offset. `XRSession.viewerSpace` returns a "viewer" reference space with no offset (for convenience). This seems conceptually clearer to me, and it keeps all the capabilities both spaces have right now.

**Comment by toji** [comment]
Nell talked to me about this briefly this morning and the idea clicked with me pretty quickly. And it turns out that OpenXR handles these as a unified concept via their `VIEW` [reference space](https://www.khronos.org/registry/OpenXR/specs/0.90/html/xrspec.html#reference-spaces). So we've got at least one example of prior art in that regard. They present it as primarily being for head-locked content, which is a sensible approach for non-screen-space uses.

I think we still need to be able to apply origin offsets to the viewer/identity space, so I'm on board with a consolidated version being an `XRReferenceSpace`. I don't think it's necessary to have two ways to get to it, though, and I don't think it's particularly intuitive to have one way to get the viewer reference space (attribute) and a different way to get all the other reference spaces (`requestReferenceSpace()`) so my suggestion in terms of IDL would be to drop `XRSession/viewerSpace` entirely and replace `identity` with `viewer` in the `XRReferenceSpaceType` enum. Feels simpler all around!

My main concerns are still privacy-based. We abssolutely need a reference space that's guaranteed to be able to be created for inline content without user consent, and that MUST NOT inadvertently leak sensor data. I think Manish's view on this is reasonable, though, in that we really just shouldn't be surfacing other spaces in that scenario anyway without additional user consent.

**TL;DR:** I'm totally open to this, and think it would represent a nice simplification.

**Comment by Manishearth** [comment]
I can write a followup PR (maybe on editor-collab?) after https://github.com/immersive-web/webxr/pull/621 lands that contains a strawman proposal and an explainer update.

I'm not sure if there's a document somewhere talking about privacy concerns but we should update that too with the reasoning I provided.

**Comment by thetuvix** [comment]
@Manishearth:
> A potential way of fixing that is to replace "identity" with a "viewer" reference space.
@toji:
> my suggestion in terms of IDL would be to drop XRSession/viewerSpace entirely and replace identity with viewer in the XRReferenceSpaceType enum. Feels simpler all around!

This approach makes great sense to me!  Apps that want to manually look around the scene relative to a dummy viewer simply manually offset from the natural view direction that you get within an inline session.

Unifying viewer space like this increases the need to get crisp on its definition, as we're discussing in https://github.com/immersive-web/webxr/pull/621#discussion_r280662081.

**Comment by toji** [comment]
Put up a pull request that covers this and #625 to better facilitate discussion. (Assuming that it will need rebasing on top of #621, but that shouldn't change it's substance much.)
